client/state/data-layer/wpcom-http/README.md

```diff
@@ -179,16 +179,22 @@ dispatch( http( {
 ```
 
 If given, the action passed as the second and optional parameter will take the place of all unspecified `onSuccess`, `onFailure`, and `onProgress` responder events.
+
 Because we have a very common pattern when issuing requests there is a built-in helper to hand each action off to the right function based on the (possibly) attached metadata.
 
 ```js
 import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
 
 // create request when no meta present, add on success, alert on failure
-dispatchRequest( fetchMenu, addMenuItems, alertFailure );
+dispatchRequest( { fetch: fetchMenu, onSuccess: addMenuItems, onError: alertFailure } );
 
 // create request when no meta present, indicate on success, undo on failure, update on progress
-dispatchRequest( sendRecipe, indicateSuccess, removeRecipe, { onProgress: updateRecipeProgress } );
+dispatchRequest( {
+	fetch: sendRecipe,
+	onSuccess: indicateSuccess,
+	onError: removeRecipe,
+	onProgress: updateRecipeProgress
+} );
 ```
 
 ### `dispatchRequest()` helper
```

Comment:
shall we temporarily renamed this to `dispatchRequestEx` as well?

Reply:
I'm not sure what to do with docs. Include documentation for both variants now? Or only the new one and leave the documentation a bit confusing for a while?

Experience says that the upgrade won't happen overnight.

Reply:
> Experience says that the upgrade won't happen overnight.
 Write

having both variants in the docs makes sense, but that sounds like too much churn for my tastes. 

Personally I would prefer we just have the `Ex` variant for now (and not the docs for the old one).  Thats because all new code should use the new variant.  Any old code that is modified is most likely going to be touched by people that are already familiar with the old code.  You could even leave a link in the docs pointing to an older commit with the old docs as a temporary measure

Reply:
I'd be happy to help convert old usages. I'm fine keeping both in for a while, but if we think we can get all the updates made in one go _and_ catch in-progress PRs then would that make it easier? just do a clean update?